src/core/qgsmaprenderersequentialjob.cpp

```diff
@@ -31,6 +31,7 @@ QgsMapRendererSequentialJob::QgsMapRendererSequentialJob( const QgsMapSettings&
   mImage = QImage( mSettings.outputSize(), mSettings.outputImageFormat() );
   mImage.setDotsPerMeterX( 1000 * settings.outputDpi() / 25.4 );
   mImage.setDotsPerMeterY( 1000 * settings.outputDpi() / 25.4 );
+  mImage.fill( settings.backgroundColor().rgba() );
```

Comment:
Ok, that makes sense. It's triggered when the fillRect is used with a partially transparent color.

In that case, I think the correct solution is to first fill mImage with a totally transparent color (eg Qt::transparent), rather then the background color. Otherwise QgsMapRendererCustomPainterJob::start() will still draw a rect over the background color, which may result in doubling the opacity of a semi-transparent background color ( eg QgsMapRendererSequentialJob::QgsMapRendererSequentialJob fills the image with (255,0,0,0.3), then QgsMapRendererCustomPainterJob::start() draws a rect over this using (255,0,0,0.3) - result won't have a background of 0.3 opacity).